s2/bq/doc/REFERENCE.md

```diff
@@ -0,0 +1,162 @@
+## Reference
+
+### S2
+
+#### VERSION
+
+{{% bannerNote type="code" %}}
+s2.VERSION()
+{{%/ bannerNote %}}
+
+Returns the current version of the s2 library.
+
+#### GEOJSONBOUNDARY_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.GEOJSONBOUNDARY_FROMKEY(key STRING) 
+{{%/ bannerNote %}}
+
+Returns a GeoJSON containing the boundary corners of a quadkey.
+
+* `key`: `STRING` quadkey to extract the corners from.
+
+#### GEOJSONBOUNDARY_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.GEOJSONBOUNDARY_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Returns a GeoJSON containing the boundary corners given a longitud, latitud and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### ID_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Convert from hilbert quadtree id to s2 cell id.
+
+* `key`: `STRING` quadkey to be converted.
+
+#### ID_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Generate the s2 cell id for a given longitud, latitud and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### KEY_FROMID
+
+{{% bannerNote type="code" %}}
+s2.KEY_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Convert from s2 cell id to hilbert quadtree id.
+
+* `id`: `INT64` s2 cell id to be converted.
+
+#### KEY_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.KEY_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Generate the quadkey for a given longitud, latitud and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### LONG_FROMID
+
+{{% bannerNote type="code" %}}
+s2.LONG_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Extract the longitude component from a give s2 cell id.
+
+* `id`: `INT64` s2 cell id.
+
+#### LONG_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.LONG_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Extract the longitude component from a quadkey.
```

Comment:
Suggest using the parlance "Hilbert quadkey" or something here to differentiate from the quadkey module.

Reply:
Actually in the quadkey module we are using majorly, quadints, which is a quadkey stored in a INT64. But yes, using HILBERTQUADKEY in this module should make it more clear.

Reply:
Just to make sure: my suggestion just pertained to documentation. The verbosity of `LONG_FROMHILBERTQUADKEY` might negatively impact function readability more than the inline context might help it, especially since this is a function developers are likely to use as an argument to other functions, ex.
`ST_GEOGPOINT(LONG_FROMHILBERTQUADKEY(columnY), LAT_FROMHILBERTQUADKEY(columnX))`